tests/rptest/services/redpanda.py

```diff
@@ -564,6 +566,9 @@ def update_rp_conf(self, conf) -> dict[str, Any]:
         if self.cloud_storage_spillover_manifest_max_segments:
             conf[
                 'cloud_storage_spillover_manifest_max_segments'] = self.cloud_storage_spillover_manifest_max_segments
+
+        conf['retention_local_strict'] = self.retention_local_strict
```

Comment:
Pretty sure this overrides the `retention_local_strict` you are setting in `log_storage_target_size_test.py`. Effectively, both test variants will have `retention_local_strict=True. You should set `retention_local_strict` via `SISettings` or override it again after the call to `set_si_settings` in the test.